torch/testing/_internal/common_methods_invocations.py

```diff
@@ -469,6 +473,23 @@ def sample_inputs(self, device, dtype, requires_grad=False):
                    handles_complex_extremals=False),
 ]
 
+if TEST_SCIPY:
+    op_db_scipy_reference = [
+        UnaryUfuncInfo('sigmoid',
+                       ref=scipy.special.expit,
+                       decorators=(precisionOverride({torch.float16: 1e-2,
+                                                      torch.bfloat16: 1e-2}),),
+                       # 'expit' not supported for the input types
```

Comment:
Is there an issue for complex cpu sigmoid support we can link here? If not, please file one and update the comment with a link to it.

Reply:
Actually the issue is that `scipy.special.expit` does not work with complex dtypes

```python
>>> import torch 
>>> from scipy.special import expit
>>> x = torch.randn(3, dtype=torch.cdouble)
>>> x.sigmoid()
tensor([0.4987-0.1376j, 0.3244+0.1567j, 0.4961+0.2211j],
       dtype=torch.complex128)
>>> expit(x.numpy())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: ufunc 'expit' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''
```

Reply:
Instead of using expit as a reference, what about using:

```
def np_sigmoid(x):
  return 1 / (1 + np.exp(-x))
```

This should be numerically stable. SciPy's expit is implemented here for comparison:

https://github.com/scipy/scipy/blob/8dba340293fe20e62e173bdf2c10ae208286692f/scipy/special/_logit.c.src#L29

And this should support complex types.

Reply:
On CPU, the test fails on extreme values.
Mostly related to https://github.com/pytorch/pytorch/issues/48010 as sigmoid uses vectorized `exp`.
https://github.com/pytorch/pytorch/blob/272f4db043ec2c63ecfe6d2759e7893cb842a3c3/aten/src/ATen/native/cpu/UnaryOpsKernel.cpp#L37-L50

Have enabled the test on `CUDA`.